user-ms/src/main/kotlin/puc/service/impl/UserAppServiceImpl.kt

```diff
@@ -87,4 +90,22 @@ class UserAppServiceImpl (
             refreshToken
         )
     }
+
+    override fun updateUser(id: Long, request: UpdateUserRequest): UserResponse {
+        val user = userRepository.findById(id).orElseThrow { UserNotFoundException(MESSAGE_ERRO_USER_NOT_FOUND) }
```

Comment:
teria que verificar se o cara possui a role de ADMIN para poder alterar ROLES. A ideia de update de usuário dei para adicionar mais campos tipo email, phone, cpfCnpj, firstName, lastName...

Reply:
por motivos de: não quero quebrar o serviço pra quem já tá usando. Adicionei essa validação em cima da DEFAULT_ROLE: ROLE_USER. quem for mexer com as roles que arrume isso